src/test/ui/issue-52717.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+enum A {
+  A {
+    foo: usize,
+  }
+}
+
+fn main() {
+  let x = A::A { foo: 3 };
+  match x {
+    A::A { fob } => { println!("{}", fob); }
```

Comment:
There's another case that we're not handling (but it's fine):

```
enum A {
  A {
    foo: usize,
  }
}
 fn main() {
  let x = A::A { foo: 3 };
  match x {
    A::A { fob } => { println!("{}", foo); }
  }
}
```

After applying the suggestion, the code will complain about the printed `fob`.

It'll be interesting coming up with a solution for this as a follow up.

Reply:
right!

Reply:
Working toward that :)